### PR TITLE
ORC-1358: Use `spotless` to format `pom` files

### PR DESCRIPTION
**Pull request by williamhyun**
### What changes were proposed in this pull request?
This PR aims to use [spotless](https://github.com/diffplug/spotless) to format pom files.

### Why are the changes needed?
This will enforce good style in pom files. 

### How was this patch tested?
Pass the CIs with newly added plugin.

### CODE REVIEW

**Comment by williamhyun** [comment]
cc: @dongjoon-hyun

File: java/bench/pom.xml [hunk 1]
```diff
@@ -117,36 +113,36 @@
...
-	    <artifactId>commons-beanutils</artifactId>
-	  </exclusion>
-	  <exclusion>
-	    <groupId>commons-beanutils</groupId>
-	    <artifactId>commons-beanutils-core</artifactId>
-	  </exclusion>
-	  <exclusion>
-	    <groupId>javax.servlet</groupId>
-	    <artifactId>servlet-api</artifactId>
-	  </exclusion>
```
Comment (dongjoon-hyun) [inline java/bench/pom.xml:140]: Nice. I saw this before in another PR, but didn't fix it at that time. Thanks.

File: java/pom.xml [hunk 2]
```diff
@@ -879,55 +365,594 @@
...
+    </profile>
+    <profile>
+      <!-- a profile to check the source for common problems -->
+      <id>analyze</id>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>com.diffplug.spotless</groupId>
+            <artifactId>spotless-maven-plugin</artifactId>
+          </plugin>
```
Comment (dongjoon-hyun) [inline java/pom.xml:910]: Thank you for putting this under `analyze` profile.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM (Pending CIs). I removed a few outdated comments.

**Comment by dongjoon-hyun** [comment]
I created a backport PR, @williamhyun .
- https://github.com/apache/orc/pull/1386

**Pull request merged**